01_install_requirements.sh

```diff
@@ -73,3 +73,47 @@ if ! which dep 2>&1 >/dev/null ; then
     mkdir -p $GOPATH/bin
     curl https://raw.githubusercontent.com/golang/dep/master/install.sh | sh
 fi
+
+# configure base directory for mirror registry
+if  [ ! -z "${MIRROR_IMAGES}" ]; then
+    # httpd-tools provides htpasswd utility
+    sudo yum install httpd-tools
+
+    sudo mkdir -pv ${REGISTRY_DIR}/{auth,certs,data}
+    sudo chown -R $USER:$USER ${REGISTRY_DIR}
+
+    pushd $REGISTRY_DIR/certs
+    SSL_HOST_NAME=$( getent hosts `hostname -s` | awk '{print $NF}' )
+
+    SSL_HOST_NAME_SHORT=$(hostname -s)
+
+    openssl genrsa -out ${REGISTRY_DIR}/certs/registry.key 2048
+
+    openssl req -x509 \
+        -key ${REGISTRY_DIR}/certs/registry.key \
+        -out  ${REGISTRY_DIR}/certs/registry.crt \
+        -days 365 \
+        -subj "/C=US/ST=NC/L=Raleigh/O=Test Company/OU=Testing/CN=${SSL_HOST_NAME}/subjectAltName=${SSL_HOST_NAME_SHORT}"
+
+    popd
+
+    htpasswd -bBc ${REGISTRY_DIR}/auth/htpasswd ${REGISTRY_USER} ${REGISTRY_PASS}
+
+    sudo cp ${REGISTRY_DIR}/certs/registry.crt /etc/pki/ca-trust/source/anchors/
+    sudo update-ca-trust
+
+    sudo podman run -d --name mirror-registry -p ${REGISTRY_PORT}:5000 \
```

Comment:
We are already starting a registry on line 44, can we replace that with this configuration?

Reply:
Updated